compiler/damlc/tests/src/DA/Test/DataDependencies.hs

```diff
@@ -1236,6 +1236,37 @@ tests Tools{damlc,repl,validate,davlDar,oldProjDar} = testGroup "Data Dependenci
             ]
         ]
 
+    , dataDependenciesTest "Using orphan instances transitively while importing a type"
+        -- This test checks that types are imported
+        -- transitively via data-dependencies, without
+        -- triggering warnings for redundant imports.
+        [
+            (,) "Type.daml"
+            [ "module Type where"
+            , "data T = T"
+            ]
+        ,   (,) "OrphanInstanceAndType.daml"
+            [ "module OrphanInstanceAndType (T(..), S(..)) where"
+            , "import Type (T(..))"
+            , "instance Show T where show T = \"T\""
+            , "data S = S"
+            , "  deriving (Show)"
+            ]
+        ,   (,) "Wrapper.daml"
+            [ "module Wrapper (T(..), S(..)) where"
```

Comment:
@cocreature this is my attempt at testing the scenario you mentioned, namely, `Main` indirectly importing both an orphan instance and a datatype from the same module. However, this fails with "`T` and `S` not in scope" since data-dependencies doesn't support reexports yet #10772 (that issue is for _module_ reexports, maybe we need a new issue?)

Reply:
good point, then let’s just drop this test for now from this PR.

Reply:
cool, will do!